functions/go/create-setters/createsetters/create_setters.go

```diff
@@ -0,0 +1,340 @@
+package createsetters
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+var _ kio.Filter = &CreateSetters{}
+
+// CreateSetters creates a comment for the resource fields which
+// contain the same value as setter value
+type CreateSetters struct {
+	// Setters holds the user provided values for simple map setters
+	Setters []Setter
+
+	// ArraySetters holds the user provided values for array setters
+	ArraySetters []ArraySetter
+
+	// Results are the results of applying setter values
+	Results []*Result
+
+	// filePath file path of resource
+	filePath string
+}
+
+// Setter stores name and value of the map setter
+type Setter struct {
+	// Name is the name of the setter
+	Name string
+
+	// Value is the input value for setter
+	Value string
+}
+
+// ArraySetter stores name and values of the array setter
+type ArraySetter struct {
+	// name of the setter
+	Name string
+
+	// values for array setter
+	Values []string
+}
+
+// Result holds result of create-setters operation
+type Result struct {
+	// FilePath is the file path of the matching value
+	FilePath string
+
+	// FieldPath is field path of the matching value
+	FieldPath string
+
+	// Value of the matching value
```

Comment:
Value is the value of the field to which setter comment is added. Update the same for other variables.